Projects/Embark/Sources/Viewables/Actions/AddressAutocompleteAction/AddressTransition.swift

```diff
@@ -0,0 +1,182 @@
+import Flow
+import Foundation
+import Presentation
+import UIKit
+import hCore
+import hCoreUI
+
+protocol AddressTransitionable {
+	var boxFrame: ReadWriteSignal<CGRect?> { get }
+}
+
+public class AddressTransition: NSObject, UIViewControllerAnimatedTransitioning {
+	let duration = 0.5
+	var presenting = true
+	let firstBox: UIView
+	let secondBox: UIView
+	let interimAddressInput: AddressInput
+	var initialBoxFrame = CGRect.zero
+
+	init(
+		firstBox: UIView,
+		secondBox: UIView,
+		addressInput: AddressInput
+	) {
+		self.firstBox = firstBox
+		self.secondBox = secondBox
+		self.interimAddressInput = addressInput
+	}
+
+	private let didEndTransitionCallbacker = Callbacker<Bool>()
+	var didEndTransitionSignal: Signal<Bool> {
+		didEndTransitionCallbacker.providedSignal
+	}
+
+	private let didStartTransitionCallbacker = Callbacker<Bool>()
+	var didStartTransitionSignal: Signal<Bool> {
+		didStartTransitionCallbacker.providedSignal
+	}
+
+	public func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
+		return duration
+	}
+
+	public func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+		let containerView = transitionContext.containerView
+		var _autocompleteView = transitionContext.view(forKey: .to)
+		if let fromView = transitionContext.view(forKey: .from), !presenting {
+			_autocompleteView = fromView
+		}
+		guard let autocompleteView = _autocompleteView else { return }
+
+		if presenting {
+			containerView.addSubview(autocompleteView)
+			containerView.bringSubviewToFront(autocompleteView)
+			autocompleteView.alpha = 0.0
+		}
+
+		firstBox.alpha = 0.0
+		secondBox.alpha = 0.0
+
+		if presenting {
+			guard let firstBoxSuperview = firstBox.superview else { return }
+			initialBoxFrame = firstBoxSuperview.convert(firstBox.frame, to: nil)
+		}
+
+		guard let secondBoxSuperview = secondBox.superview else { return }
+		var destinationFrame = secondBoxSuperview.convert(secondBox.frame, to: nil)
+		if presenting {
+			// Hack. autocompleteView should layout first, then we get the correct y position, but then animations are not working.
```

Comment:
could you do `setNeedsLayout` & `layoutIfNeeded` maybe?

Reply:
True, it worked!!!